common/pooll2tx.go

```diff
@@ -92,14 +92,17 @@ func NewPoolL2Tx(tx *PoolL2Tx) (*PoolL2Tx, error) {
 
 // SetType sets the type of the transaction
 func (tx *PoolL2Tx) SetType() error {
-	if tx.ToIdx >= IdxUserThreshold {
+	isAddrEmpty := tx.ToBJJ == EmptyBJJComp && tx.ToEthAddr == EmptyAddr
+	if isAddrEmpty && tx.ToIdx >= IdxUserThreshold {
```

Comment:
Have you verified that hermezjs sets the ToEthAddr = 0 && ToBjj = 0 when doing a TxTypeTransfer?  I'm asking because protocol wise, a tx with
- ToIdx >= IdxUserThreshold
- ToEthAddr != 0
- ToBjj != 0
is valid.  I think it's fine to reject these txs in the implementation, but I want to make sure that hermezjs doesn't send them like this.

Reply:
let me check into the hermezjs, also will check with the front end team as well

Reply:
These are the requests made from the wallet UI:

Making a `TransferToBJJ`:
```json
{
  "type": "TransferToBJJ",
  "tokenId": 0,
  "fromAccountIndex": "hez:ETH:1276",
  "toAccountIndex": null,
  "toHezEthereumAddress": "hez:0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF",
  "toBjj": "hez:9Ez-tUR6uRSep2kEqsl3ENeeo9AYw6gr9qgcUVHsIqHL",
  "amount": "100000000000000",
  "fee": 194,
  "nonce": 33,
  "requestFromAccountIndex": null,
  "requestToAccountIndex": null,
  "requestToHezEthereumAddress": null,
  "requestToBjj": null,
  "requestTokenId": null,
  "requestAmount": null,
  "requestFee": null,
  "requestNonce": null,
  "id": "0x025a30c274aced55a6c14f000344254d3cb91b305d08e964366c5c7db34b8ae1d6",
  "signature": "e7ec02d5bbd8ce0ae76aa3f512c19f3c9fbd47f33847ceb6a0e8bb8a139e439276697052c81612701530ee4f3e5661dbeb2fc38fa3096a4794bc9294a0c80905"
}
```

and making a transfer to an ETH Hez address, the wallet gets the IDX from the API to make a regular `Transfer`
```json
{
  "type": "Transfer",
  "tokenId": 0,
  "fromAccountIndex": "hez:ETH:1276",
  "toAccountIndex": "hez:ETH:1269",
  "toHezEthereumAddress": null,
  "toBjj": null,
  "amount": "100000000000000",
  "fee": 194,
  "nonce": 34,
  "requestFromAccountIndex": null,
  "requestToAccountIndex": null,
  "requestToHezEthereumAddress": null,
  "requestToBjj": null,
  "requestTokenId": null,
  "requestAmount": null,
  "requestFee": null,
  "requestNonce": null,
  "id": "0x0252222e0de00610eb3d32a3cbe72020ab8810c3518140e3bf82f1be663b26f1d2",
  "signature": "c0b83434ada09c04f9a99baf104c570b95d6f376409f30f9667a770218e4799cbf666a6f924b362ccce08ef8a5dbe32635751c2c3500039b78ef6b4e27be1101"
}
```

Reply:
<img width="1431" alt="Screen Shot 2021-04-15 at 17 38 24" src="https://user-images.githubusercontent.com/2406457/114935335-7042f100-9e11-11eb-9d97-9e8659735971.png">